### ISSUE TITLE
TypeError instead of RuntimeError

### ISSUE THREAD

**Issue by mjwen** [mjwen/kimpy#10]
Hi @yafshar , I built kimpy with GCC 10.2.0 on macOS and then ran the tests locally. Instead of RuntimeError, I got TypeError. Any idea? 

```
(kliff) mjwen@Mingjian-MBP ~/A/k/tests (devel)> pwd
/Users/mjwen/Applications/kimpy/tests
(kliff) mjwen@Mingjian-MBP ~/A/k/tests (devel)> python test_charge_unit.py
Traceback (most recent call last):
  File "test_charge_unit.py", line 59, in <module>
    test_main()
  File "test_charge_unit.py", line 51, in test_main
    inst = kimpy.charge_unit.get_charge_unit(N)
TypeError: get_charge_unit(): incompatible function arguments. The following argument types are supported:
    1. (index: int) -> kimpy.charge_unit.ChargeUnit

Invoked with: 4
```

**Comment by yafshar** [comment]
@mjwen sounds a bit strange. Please check if you have compiled KIM-API with the same compiler.

I have tested kimpy on macOS BigSUR `11.2.3 (20D91)` with 

`g++-10 (Homebrew GCC 10.2.0_4) 10.2.0` 
`Python 3.9.4`
KIM-API -> `2.2.1+v2.2.1.GNU.GNU.GNU`

as well as 

`Apple clang version 12.0.0 (clang-1200.0.32.29)` 
`Python 3.9.4`
KIM-API -> `2.2.1+v2.2.1.AppleClang.AppleClang.GNU`

and 

`icpc (ICC) 2021.2.0 20210228`
`Python 3.7.9 :: Intel Corporation`
KIM-API -> `2.2.1+v2.2.1.Intel.Intel.Intel`

**Comment by yafshar** [comment]
within the binding, it only throws a runtime error

```cpp
    int error = CHARGE_UNIT::GetChargeUnit(index, &charge_unit);
    if (error == 1)
    {
      throw std::runtime_error("index < 0 or index >= numberOfChargeUnits!");
    }
```

**Comment by yafshar** [comment]
@mjwen I updated the GitHub actions on macOS to use the GCC 10 https://github.com/openkim/kimpy/commit/41d23b9722dba0daae20f830983b266000b47522 and it is working correctly.

**Comment by yafshar** [comment]
@mjwen I close this issue for now. If there is any issue please reopen with more detailed info.

**Comment by mjwen** [comment]
@yafshar After a fair amount of debugging, I finally figured it out. 

Everything is working like a charm, no bugs there :) This is reassuring ! 

The reason I saw the TypeError is because I messed up my Python environment to make brew gcc work with command line tools. I recently upgraded my macOS to version 11.2.3, but the brew gcc will installed with version 10.x. I did some trick to make brew gcc be able to find the headers used to be in `/user/include` but now in `/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/`. This trick somehow breaks my Python environments, which leads to the TypeError (see https://github.com/pybind/pybind11/issues/1829)

To fix this, I 
- delete and reinstall command line tools 
- delete and reinstall brew gcc 

Then everything works.
